lib/src/main/scala/spinal/lib/bus/regif/BusIfAdapter/AhbLite3BusInterface.scala

```diff
@@ -18,21 +18,20 @@ case class AhbLite3BusInterface(bus: AhbLite3, sizeMap: SizeMapping, regPre: Str
 
   val askWrite    = bus.HSEL & bus.HTRANS(1) === True &  bus.HWRITE
   val askRead     = bus.HSEL & bus.HTRANS(1) === True & !bus.HWRITE
-  val doWrite     = bus.HREADY & RegNext(askWrite, False)
-  val doRead      = bus.HREADY & askRead
+  val doWrite     = bus.HREADYOUT & RegNext(askWrite, False)
+  val doRead      = bus.HREADYOUT & askRead
 
   val addressDelay = RegNextWhen(bus.HADDR, askRead | askWrite)
-  readError.clearWhen(askWrite)
 
   bus.HREADYOUT := True
-  bus.HRESP     := False
+  bus.HRESP     := readError
```

Comment:
Depending on how `readError` is driven, this implementation might not match the protocol.

Reply:
You removed the line for `HREADYOUT`, I guess because of the `otherwise` line 38, but kept the one for HRESP. I don't understand.